WISE-Community/WISE-Client
# Builds in other languages

## Changes
- Production builds now output in different folders.
- SiteStyles.[hash].css is copied to SiteStyles.css in en-US. This is a fix for the TinyMCE css reference
- Development server can be run with another language, with ```--configuration=zh-Hant``` option in ```ng server ...``` command

## Test Development build

1. Change your browser's language settings to Simplified Chinese.
2. Test local development: in package.json by changing the line
```ng serve --hmr --host 0.0.0.0"```
to
```ng serve --configuration=zh-Hant --hmr --host 0.0.0.0"```
3. Restart docker wise-client ```docker restart wise-client```
4. Verify that the local server is now running a Simplified Chinese build of WISE.
5. Test that everything works as before in the Simplified Chinese interface.

## Test Production build
Putting this on production server will require changes to the Nginx conf file. We can do that together once this PR is approved. (Don't merge this PR until we test on production together)

0. Make sure that Docker has enough resources. I have allocated 10GB of memory. (https://docs.docker.com/docker-for-windows/)
1. Make sure that wise-client docker container is running.
2. Build a production build of WISE-Client
```docker exec -it wise-client npm run build-prod```
3. This will take anywhere from 10-20 minutes, depending on your computer speed. It will output the builds in the dist folder.
4. Verify that in the dist folder, you have folders: en-US, es, ja, tr, pt, zh-Hant, zh-Hans
5. Verify that dist/en-US/index.html's base href is "/".
6. Verify that dist/[lang]/index.html's base href is "/[lang]/" for other locales.
7. Verify that dist/en-US contains both siteStyles.[hash].css and siteStyles.css.

## Set up production server locally
1. Download NginX 
2. In nginx.conf, add this line inside the http block:
```
http {
...
    # Browser preferred language detection (does NOT require AcceptLanguageModule)
    map $http_accept_language $accept_language {
        ~*^de de;
        ~*^fr fr;
        ~*^en en;
        ~*^ja ja;
        ~*^zh-Hans zh-Hans;
        ~*^zh-Hant zh-Hant;
        ~*^zh-CN zh-Hans;
        ~*^zh-TW zh-Hant;
    }
...
}
```
3.  Add wise.conf to sites-enabled and make sure it’s loaded by Nginx. You will need to edit the path to the dist folder and maybe the port number based on your own Nginx port.
4. Make sure that wise-api Docker container is running (should run on port 8080 by default)
5. Start nginx and go to localhost:82 (could be different port for you)
6. Test everything out and make sure it works

Closes #92
